### PR TITLE
Make scrumlr a PWA

### PR DESCRIPTION
**Pull request by coehne**
## Description
I read the #1734 issue and took it as an opportunity to dive deeper into the CRA implementation of service workers.
The code passes the lighthouse test for PWA and can be installed as one (I tested it on macOS).
The service worker is only registered on the production build so that development will still work without worrying about the caching.

The service worker will only update the app for the client once all local tabs in the browser are closed. This could be considered as a downside. As an alternative, I could think about implementing a new service-worker, not using the implementation of CRA and enforce a NetworkOnly caching policy.

I also added [maskable icons](https://web.dev/maskable-icon/) which are used for the new Android theming.
Is there a figma/XD design where I could find the correct icons? I improvised with the logo which was inside the assets folder.

## Changelog

- Installed 3 workbox npm packages plus 3 types packages as dev dependencies
- Imported src/service-worker.ts and src/serviceWorkerRegistration.ts from [CRA-PWA-TS template](https://github.com/cra-template/pwa) and adjusted them
- Added service worker registration in index.tsx
- Added maskable logos and a manifest.json for PWA installations
- Changed index.html to include apple-touch-icon and the manifest.json file

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The light- and dark-theme are both supported and tested

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -2,16 +2,15 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <link href="https://static.inovex.de/css/raleway.css" rel="stylesheet">
+    <link href="https://static.inovex.de/css/raleway.css" rel="stylesheet" />
```
Comment (bitionaire) [inline public/index.html:5]:
```suggestion
```

We removed the usage of the inovex CDN so this line is no longer necessary

File: public/index.html [hunk 2]
```diff
@@ -2,16 +2,15 @@
...
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <meta name="theme-color" content="#000000" />
-    <meta
-      name="description"
-      content="Open source online retrospective & collaboration tool - fast onboarding, jump right into a session with your team"
-    />
+    <meta name="description" content="Open source online retrospective & collaboration tool - fast onboarding, jump right into a session with your team" />
     <meta name="keywords" content="Collaboration, Scrum, Retrospective, Retro, Agile" />
     <title>scrumlr.io - Online collaboration</title>
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (bitionaire) [inline public/index.html:12]: I think we should support touch icons for different devices and resolutions here. See https://www.web-design-und-so.de/webdesign/touch-icons-apple-android-windows/ this link for more information on this topic. Basically a complete list would look something like this:

```
<link rel="shortcut icon" href="%PUBLIC_URL%/logo196.png" sizes="196x196">
<link rel="shortcut icon" href="%PUBLIC_URL%/logo32.png" sizes="32x32">
<link rel="shortcut icon" href="%PUBLIC_URL%/logo16.png" sizes="16x16">

<link rel="apple-touch-icon" href="%PUBLIC_URL%/apple-touch-icon-180.png">
<link rel="apple-touch-icon" href="%PUBLIC_URL%/apple-touch-icon-167.png" sizes="167x167"> 
<link rel="apple-touch-icon" href="%PUBLIC_URL%/apple-touch-icon-152.png" sizes="152x152"> 

<meta name="msapplication-square70x70logo" content="%PUBLIC_URL%/logo70.png">
<meta name="msapplication-square150x150logo" content="%PUBLIC_URL%/logo150.png">
<meta name="msapplication-wide310x150logo" content="%PUBLIC_URL%/logo310x150.png">
<meta name="msapplication-square310x310logo" content="%PUBLIC_URL%/logo310.png">
```

I can ask  our design team to provide you with the resources.

File: public/manifest.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "short_name": "Scrumlr",
```
Comment (bitionaire) [inline public/manifest.json:2]:
```suggestion
  "short_name": "scrumlr.io",
```
I prefer the lowercase title with our TLD included. Just personal preference :)

**Review by bitionaire — COMMENTED** [review]
The included logo's have an issue with the missing gap within the "L" letter. I will ask our design team whether they can provide better logos with broader support for different resolutions. 
Rest of the code looks good to me. Thank you @Dakicka for your support and your contribution to this project!

**Review by bitionaire — APPROVED** [review]

**Review by brandstetterm — APPROVED** [review]

**Pull request merged**